c_emulator/riscv_sim.cpp

```diff
@@ -63,14 +62,7 @@ FILE *trace_log = NULL;
 char *dtb_file = NULL;
 unsigned char *dtb = NULL;
 size_t dtb_len = 0;
-#ifdef RVFI_DII
-static bool rvfi_dii = false;
-/* Needs to be global to avoid the needed for a set-version packet on each trace
- */
-static unsigned rvfi_trace_version = 1;
-static int rvfi_dii_port;
-static int rvfi_dii_sock;
-#endif
+rvfi_t *rvfi = NULL;
```

Comment:
```suggestion
std::optional<rvfi_t> rvfi;
```
Avoids the need for heap allocation and free.